src/vs/nls.ts

```diff
@@ -155,13 +155,49 @@ export function load(name: string, req: AMDLoader.IRelativeRequire, load: AMDLoa
 					messagesLoaded(messages);
 				}
 			});
+		} else if (pluginConfig.translationServiceUrl) {
+			if (language === null) {
+				return req([name + '.nls'], messagesLoaded);
+			}
+			const url = endWithSlash(pluginConfig.translationServiceUrl) + endWithSlash(language) + 'vscode/' + endWithSlash(name);
+			(async () => {
+				try {
+					const res = await fetch(url);
+					if (res.ok) {
+						const messages = await res.json();
+						return messagesLoaded(messages);
+					}
+					throw new Error(`${res.status} - ${res.statusText}`);
+				} catch (err) {
+					if (!language.includes('-')) {
+						return req([name + '.nls'], messagesLoaded);
+					}
+					const url = endWithSlash(pluginConfig.translationServiceUrl!) + endWithSlash(language.split('-')[0]) + 'vscode/' + endWithSlash(name);
+					try {
+						const res2 = await fetch(url);
+						if (res2.ok) {
+							const messages = await res2.json();
+							return messagesLoaded(messages);
+						}
+						throw new Error(`${res2.status} - ${res2.statusText}`);
+					} catch (err) {
+						return req([name + '.nls'], messagesLoaded);
+					}
```

Comment:
As of now, languages are either `de` or `de-FOO`. There is no `de-FOO-BAR` so if there's a `-` we grab the first part which should be the root language.